### PR TITLE
Flash: Propagate payment failure down the path

### PR DESCRIPTION
**Pull request by omerfirmak**
If a node on the path fails to route the payment
the already created HTLCs should unwind.

Part of #1623

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
On mobile it LGTM. I'll have another look soon.

**Comment by AndrejMitrovic** [comment]
Had another look, LGTM!

**Comment by omerfirmak** [comment]
Rebased & added some logs

File: source/agora/test/Flash.d [hunk 1]
```diff
@@ -576,6 +576,13 @@ unittest
     writefln("Sleeping for 7 seconds..");
     Thread.sleep(7.seconds);
 
+    // Second payment attempt should fail because of insufficient funds
+    auto inv_2 = charlie.createNewInvoice(Amount(4_000), time_t.max, "payment 2");
+    alice.payInvoice(inv_2);
+
+    writefln("Sleeping for 7 seconds..");
+    Thread.sleep(7.seconds);
```
Comment (AndrejMitrovic) [inline source/agora/test/Flash.d]: payInvoice is now blocking so I think we can get rid of the sleep calls between the payInvoice calls in this second test. I did get rid of them in the first test but I forgot about the second one.

Comment (omerfirmak) [reply source/agora/test/Flash.d]: Done

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**